test/testkit/matchers/trace/trace_matchers_test.go

```diff
@@ -1,114 +1,104 @@
+			"response_size":           "31",
+			"upstream_cluster.name":   "inbound|4317||",
+			"istio.canonical_service": "backend",
+		},
+	},
+	{
+		Name: "ingress-2",
+		ResourceAttributes: map[string]string{
+			"service.name":        "monitoring-custom-metrics",
+			"k8s.pod.ip":          "10.42.1.73",
+			"k8s.deployment.name": "istio",
+		},
+		SpanAttributes: map[string]string{
+			"response_size":           "32",
+			"upstream_cluster.name":   "inbound|4318||",
+			"istio.canonical_service": "istio",
+		},
+	},
+}
+
+var _ = Describe("HaveFlatTraces", func() {
 	It("should apply matcher to valid trace data", func() {
 		td := ptrace.NewTraces()
-		Expect(mustMarshalTraces(td)).Should(WithTds(ContainElements()))
+		Expect(mustMarshalTraces(td)).Should(HaveFlatTraces(ContainElements()))
 	})
 
 	It("should fail when given empty byte slice", func() {
-		Expect([]byte{}).Should(WithTds(BeEmpty()))
+		Expect([]byte{}).Should(HaveFlatTraces(BeEmpty()))
 	})
 
 	It("should return error for nil input", func() {
-		success, err := WithTds(BeEmpty()).Match(nil)
+		success, err := HaveFlatTraces(BeEmpty()).Match(nil)
 		Expect(err).Should(HaveOccurred())
 		Expect(success).Should(BeFalse())
 	})
 
 	It("should return error for invalid input type", func() {
-		success, err := WithTds(BeEmpty()).Match(struct{}{})
+		success, err := HaveFlatTraces(BeEmpty()).Match(struct{}{})
 		Expect(err).Should(HaveOccurred())
 		Expect(success).Should(BeFalse())
 	})
-})
-
-var _ = Describe("WithResourceAttrs", func() {
-	It("should apply matcher", func() {
-		td := ptrace.NewTraces()
-		rm := td.ResourceSpans().AppendEmpty()
-		attrs := rm.Resource().Attributes()
-		attrs.PutStr("k8s.cluster.name", "cluster-01")
-		attrs.PutStr("k8s.deployment.name", "nginx")
-
-		Expect(mustMarshalTraces(td)).Should(ContainTd(WithResourceAttrs(ContainElement(HaveKey("k8s.cluster.name")))))
-	})
-})
-
-var _ = Describe("WithSpans", func() {
-	It("should apply matcher", func() {
-		td := ptrace.NewTraces()
-		rs := td.ResourceSpans().AppendEmpty()
-		spans := rs.ScopeSpans().AppendEmpty().Spans()
-		spans.AppendEmpty()
-		spans.AppendEmpty()
-
-		Expect(mustMarshalTraces(td)).Should(ContainTd(WithSpans(HaveLen(2))))
-	})
-})
 
-var _ = Describe("WithTraceID", func() {
-	It("should apply matcher", func() {
+	It("should return a FlatTrace struct", func() {
 		td := ptrace.NewTraces()
-		rs := td.ResourceSpans().AppendEmpty()
-		spans := rs.ScopeSpans().AppendEmpty().Spans()
-
-		traceID := newTraceID()
-		spans.AppendEmpty().SetTraceID(traceID)
-
-		Expect(mustMarshalTraces(td)).Should(ContainTd(ContainSpan(WithTraceID(Equal(traceID)))))
+		//set resource attributes
+		rt := td.ResourceSpans().AppendEmpty()
+		attrs := rt.Resource().Attributes()
+		attrs.PutStr("service.name", "backend")
+		attrs.PutStr("k8s.pod.ip", "10.42.1.76")
+		attrs.PutStr("k8s.deployment.name", "backend")
+		//set scope version, name and attributes
+		scope := rt.ScopeSpans().AppendEmpty()
```

Comment:
Do we need to set scope name and scope version?
I see that the `FlatTrace` doesn't have fields for these values